pinot-common/src/main/java/org/apache/pinot/common/messages/SegmentReloadMessage.java

```diff
@@ -35,6 +35,9 @@ public class SegmentReloadMessage extends Message {
 
   private static final String FORCE_DOWNLOAD_KEY = "forceDownload";
 
+  // todo (saurabh) : getMsgId() returns different id on the server side than the one being set at controller. Check
```

Comment:
Interesting. The msgId should just be the ZNode id

Reply:
Yeah the getMsgId does fetch the record's zk id but for some reason I was seeing this difference on server and controller side. I'll look into this more and update the PR accordingly if we are able to use msgId itself.

Reply:
Any update on this?

Reply:
So helix, when sending a PARTICIPANT message, generates one message per partition to store into ZK. When doing that, it overrides the original message's id with a new random one
https://github.com/apache/helix/blob/master/helix-core/src/main/java/org/apache/helix/messaging/DefaultMessagingService.java#L199

It only retains the fields from the original message